src/tools/transactions_builder.rs

```diff
@@ -0,0 +1,255 @@
+            .push((action.clone(), password, BlobIndex(self.blobs.len())));
+        self.blobs.push(action.as_blob("hydentity".into()));
+    }
+    fn add_hyllar_cf(
+        &mut self,
+        token: ContractName,
+        action: ERC20Action,
+        caller: Option<BlobIndex>,
+    ) {
+        self.hyllar_cf
+            .push((action.clone(), token.clone(), BlobIndex(self.blobs.len())));
+        self.blobs.push(action.as_blob(token, caller, None));
+    }
+    fn add_stake_cf(&mut self, action: StakingAction) {
+        self.staking_cf
+            .push((action.clone(), BlobIndex(self.blobs.len())));
+        self.blobs
+            .push(action.as_blob("staking".into(), None, None));
+    }
+
+    pub async fn verify_identity(
+        &mut self,
+        state: &Hydentity,
+        password: String,
+    ) -> Result<(), Error> {
+        let nonce = get_nonce(state, &self.identity.0).await?;
+        let password = Password(BlobData(password.into_bytes().to_vec()));
+
+        self.add_hydentity_cf(
+            IdentityAction::VerifyIdentity {
+                account: self.identity.0.clone(),
+                nonce,
+            },
+            password,
+        );
+
+        Ok(())
+    }
+
+    pub fn register_identity(&mut self, password: String) {
+        let password = Password(BlobData(password.into_bytes().to_vec()));
+
+        self.add_hydentity_cf(
+            IdentityAction::RegisterIdentity {
+                account: self.identity.0.clone(),
+            },
+            password,
+        );
+    }
+
+    pub fn approve(&mut self, token: ContractName, spender: String, amount: u128) {
+        self.add_hyllar_cf(token, ERC20Action::Approve { spender, amount }, None);
+    }
+
+    pub fn transfer(&mut self, token: ContractName, recipient: String, amount: u128) {
+        self.add_hyllar_cf(token, ERC20Action::Transfer { recipient, amount }, None);
+    }
+
+    pub fn stake(
+        &mut self,
+        token: ContractName,
+        staking_contract: ContractName,
+        amount: u128,
+    ) -> Result<(), Error> {
+        self.add_stake_cf(StakingAction::Stake { amount });
+        self.add_hyllar_cf(
+            token,
+            ERC20Action::Transfer {
+                recipient: staking_contract.0.clone(),
+                amount,
+            },
+            None,
+        );
+
+        Ok(())
+    }
+
+    pub fn delegate(&mut self, validator: ValidatorPublicKey) -> Result<(), Error> {
+        self.add_stake_cf(StakingAction::Delegate { validator });
+
+        Ok(())
+    }
+
+    pub async fn build(&mut self, states: &mut States) -> Result<BuildResult> {
+        let mut new_states = states.clone();
+        let mut outputs = vec![];
+        for id in self.hydentity_cf.iter() {
+            let runner = ContractRunner::new(
+                "hydentity".into(),
+                get_binary("hydentity".into())?,
+                self.identity.clone(),
+                id.1 .0.clone(),
```

Comment:
J'aurais dit que y'avait une type 
```suggestion
                id.1.0.clone(),
```
mais wtf pq rust-analyzer veut un esapce ici ? Oo